mslib/msui/editor.py

```diff
@@ -284,15 +336,20 @@ def __init__(self, parent=None):
         self.view.setAlternatingRowColors(True)
         self.view.setColumnWidth(0, self.view.width() // 2)
 
-    def get_root_index(self, index, parents=False):
-        parent_list = []
-        while index.parent().isValid():
-            index = index.parent()
-            parent_list.append(index)
-        parent_list.reverse()
-        if parents:
-            return index, parent_list
-        return index
+        # Add invalidity roles and update status of keys
+        self.update_view()
+
+    def dict_raise_on_duplicates_empty(self, ordered_pairs):
+        """Reject duplicate and empty keys."""
+        d = {}
```

Comment:
d is not a good name. I would use "result" to indicate from the beginning that "d" will be returned.
k, v is borderline okay, but key, value would be more apparent and not that much more to type either.

Reply:
I'll update them.